lib/Doctrine/Common/Cache/FileCache.php

```diff
@@ -78,8 +109,60 @@ public function __construct($directory, $extension = '')
             ));
         }
 
+        $extension = (string) $extension;
         $this->directory = realpath($directory);
-        $this->extension = (string) $extension;
+        $this->extension = $extension ?: $this->extension;
+        $this->fileMode = 0666 & ~umask();
+    }
+
+    /**
+     * Sets the mode that new directories will be created with.
+     * @param int $mode Mode, normally in octal (ex: 0755)
+     */
+    public function setDirectoryMode($mode)
```

Comment:
and cache files are not executable. Look at the current code. It uses ``0777`` as base for folders but ``0666`` for files